--- a/net/minecraft/village/VillageSiege.java
+++ b/net/minecraft/village/VillageSiege.java
@@ -11,6 +11,7 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.GameRules;
 import net.minecraft.world.ServerWorld;
 import net.minecraft.world.gen.Heightmap;
 import net.minecraft.world.spawner.WorldEntitySpawner;
@@ -33,7 +34,7 @@
 
     public void tick()
     {
-        if (this.world.isDaytime())
+        if (this.world.isDaytime() || this.world.getGameRules().getBoolean(GameRules.DISABLE_ZOMBIE_SIEGE))
         {
             this.siegeState = VillageSiege.State.SIEGE_DONE;
             this.hasSetupSiege = false;
