--- a/net/minecraft/village/VillageSiege.java
+++ b/net/minecraft/village/VillageSiege.java
@@ -14,7 +14,9 @@
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.gen.Heightmap;
 import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.GameRules;
 
+
 public class VillageSiege
 {
     private boolean hasSetupSiege;
@@ -27,7 +29,7 @@
 
     public int func_225477_a(ServerWorld p_225477_1_, boolean p_225477_2_, boolean p_225477_3_)
     {
-        if (!p_225477_1_.isDaytime() && p_225477_2_)
+        if (!p_225477_1_.isDaytime() && p_225477_2_ && !p_225477_1_.getGameRules().getBoolean(GameRules.DISABLE_ZOMBIE_SIEGE))
         {
             float f = p_225477_1_.getCelestialAngle(0.0F);
 
