--- a/net/minecraft/world/GameRules.java
+++ b/net/minecraft/world/GameRules.java
@@ -79,6 +79,7 @@
     public static final GameRules.RuleKey<GameRules.BooleanValue> FIRE_DAMAGE = register("fireDamage", GameRules.BooleanValue.create(true));
     public static final GameRules.RuleKey<GameRules.BooleanValue> field_230127_D_ = register("doPatrolSpawning", GameRules.BooleanValue.create(true));
     public static final GameRules.RuleKey<GameRules.BooleanValue> field_230128_E_ = register("doTraderSpawning", GameRules.BooleanValue.create(true));
+    public static final GameRules.RuleKey<GameRules.BooleanValue> DISABLE_ZOMBIE_SIEGE = register("disableZombieSiege", GameRules.BooleanValue.create(true));
     private final Map < GameRules.RuleKey<?>, GameRules.RuleValue<? >> rules = GAME_RULES.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, (p_226684_0_) ->
     {
         return p_226684_0_.getValue().createValue();
