--- a/net/minecraft/world/ServerWorld.java
+++ b/net/minecraft/world/ServerWorld.java
@@ -305,9 +305,9 @@
             this.getWorldInfo().setDifficulty(Difficulty.HARD);
         }
 
-        if (this.allPlayersSleeping && this.players.stream().noneMatch((p_217449_0_) ->
+        if (this.allPlayersSleeping && this.players.stream().anyMatch((p_217449_0_) ->
     {
-        return !p_217449_0_.isSpectator() && !p_217449_0_.isPlayerFullyAsleep();
+        return !p_217449_0_.isSpectator() && p_217449_0_.isPlayerFullyAsleep();
         }))
         {
             this.allPlayersSleeping = false;
@@ -601,7 +601,7 @@
                 }
             }
 
-            this.allPlayersSleeping = j > 0 && j >= this.players.size() - i;
+            this.allPlayersSleeping = (j > 0);
         }
     }
 
